common/determined_common/experimental/experiment.py

```diff
@@ -16,9 +20,75 @@ class is obtained via :class:`~determined.experimental.Determined`, the
             master URL is automatically passed into this constructor.
     """
 
-    def __init__(self, experiment_id: int, master: str):
-        self.id = experiment_id
-        self._master = master
+    def __init__(self, api_client, master, experiment_data=None):
+        self.master = master
+        self.api_client = api_client
+
+        self.id = None
+        self.state = None
+        self.config = None
+
+        for attribute in experiment_data:
+            setattr(self, attribute, experiment_data[attribute])
+
+        self.metric = self.config.get("searcher").get("metric")
+        self.smaller_is_better = self.config.get("searcher").get("smaller_is_better")
+
+    @classmethod
+    def create_experiment(
+        cls, api_client, config, context_path, local=False, test=False, master=""
+    ):
```

Comment:
I think this makes much more sense as logic in the `Determined.create_experiment()`